modules/ROOT/pages/dataweave-functions.adoc

```diff
@@ -64,6 +64,44 @@ toPhoneFormat("1234567890")
 "(123) 456-7890"
 ----
 
+=== Overloading Functions
+
+DataWeave allows you to create multiple functions with the same name but different parameters. This is useful for defining different behaviors based on the arguments of a function call.
+
+The difference in parameters could be based on the number of parameters or the types of those parameters. To understand more about defining type constraints on function parameters you can check the xref:dataweave-type-system.adoc[type system] docs.
+
+DataWeave uses the first function it finds that accepts the arguments of the function call, in the order that they are declared in the script.
+
+.Example: Function that Uses Function Overloading
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/json
+
+fun toUpper(a: String) = upper(a)
+fun toUpper(a: Any) = null
+fun toUpper(a: String, b: Number) = upper(a) ++ b as String
+---
+toUpper("hi!")
+----
+
+The argument of the function call `toUpper("hi!")` matches with the types String and Any (so it could be called with the first two definitions) but DataWeave will execute the function `fun toUpper(a: String) = upper(a)` because is defined before the one that uses the type Any.
```

Comment:
```suggestion
The argument of the function call `toUpper("hi!")` matches the types String and Any, so it is possible to call the function with the first two definitions. However, DataWeave executes the function `fun toUpper(a: String) = upper(a)` because that function is defined before the one that uses the type Any.
```